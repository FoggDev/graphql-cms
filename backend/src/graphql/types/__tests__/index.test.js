import typeDefs from '../index'

describe('#typeDefs', () => {
  it('should match the GraphQL schema', () => {
    expect(typeDefs).toMatchSnapshot()
  })
})
