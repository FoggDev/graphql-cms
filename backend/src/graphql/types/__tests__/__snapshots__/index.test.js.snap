// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#typeDefs should match the GraphQL schema 1`] = `
"type Post {
  id: UUID!
  userId: UUID!
  title: String!
  slug: String!
  content: String!
  readingTime: String!
  language: String!
  image: String!
  published: Boolean!
  createdAt: Datetime!
  updatedAt: Datetime!
  tags: [Tag!]
}

type Query {
  getPosts: [Post!]
  getUsers: [User!]
}

type Mutation {
  createPost(input: CreatePostInput): Post!
  createUser(input: CreateUserInput): User!
  login(input: LoginInput): AuthPayload!
}

input CreatePostInput {
  title: String!
  slug: String!
  content: String!
  userId: UUID!
  tags: [TagsInput]
}

scalar UUID

scalar Datetime

type Tag {
  id: UUID!
  name: String!
}

input TagsInput {
  name: String!
}

type User {
  id: UUID!
  username: String!
  password: String!
  email: String!
  privilege: String!
  active: Boolean!
  createdAt: Datetime!
  updatedAt: Datetime!
  posts: [Post!]
}

input CreateUserInput {
  username: String!
  password: String!
  email: String!
}

input LoginInput {
  email: String!
  password: String!
}

type AuthPayload {
  token: String!
}

schema {
  query: Query
  mutation: Mutation
}
"
`;
